lib/auth/auth.js

```diff
@@ -100,7 +100,9 @@ function extractParams(request, log, awsService, data) {
  * @param {object} log - Logger object
  * @param {function} cb - the callback
  * @param {string} awsService - Aws service related
- * @param {RequestContext[]} requestContexts - array of RequestContext
+ * @param {RequestContext[] | null} requestContexts - array of
```

Comment:
Sending an empty array results in validation problems in vault since we are using the querystring to send authentication request info to vault.  
> querystring.stringify({rc: []})
''

Also, for chunk authentication requests in streaming v4 auth, we are also sending vault null for the request context.  So, by sending null here we limit the options vault has to deal with --either null or an array of requestContext instances (rather than null, an array of requestContext instances or an empty string key once querystring.parse does its magic).